Update ci to f34

AdamSaleh opened the pull request:
Update vagrant env and Dockerfiles to use f34 and make changes to CI and mergify

humaton committed:
Use f34 in vagrant environment

Signed-off-by: Tomas Hrcka <thrcka@redhat.com>

humaton committed:
Replace f32 Dockerfile with 34

Signed-off-by: Tomas Hrcka <thrcka@redhat.com>

AdamSaleh committed:
Change cico.pipeline and mergify config to take f34 into account

Signed-off-by: Adam Saleh <asaleh@redhat.com>

AdamSaleh committed:
Add changes from pull #4226

Signed-off-by: Adam Saleh <asaleh@redhat.com>

AdamSaleh committed:
Revert "Add allReleases query to the GraphQL endpoint. This commit adds a new query allReleases that allows to fetch all Release objects from the database using GraphQL. Fixes #4024."

This reverts commit 84fdcb4eed4d4f5847c4ff68666ced6c453bd758.

Signed-off-by: Adam Saleh <asaleh@redhat.com>

AdamSaleh committed:
Revert "added graphql service"

This reverts commit fe46499fa6ae2871b888f7dfe03bf74bf9ec1225.

Signed-off-by: Adam Saleh <asaleh@redhat.com>

LenkaSeg commented on devel/ci/cico.pipeline:
```diff
@@ -161,16 +161,16 @@ node('bodhi') {
             parallel(
-                f32: {test_release('f32')},
-                f33: {test_release('f33')},
+                f33: {test_release('f32')},
```
Shouldn't it be 
`f33: {test_release('f33')},
f34: {test_release('f34')},`
?

LenkaSeg replied:
Proposed changes:
https://github.com/LenkaSeg/bodhi/commit/93b432c9da6f19beaf61be4cc50b94fc4eb7a814

LenkaSeg committed:
Move -A celery option, fix typo

Signed-off-by: Adam Saleh <asaleh@redhat.com>

AdamSaleh committed:
Various fixes to make the CI pass

Fix test failure with unescaped sphinx version pin
Disable repo that fails with rawhide integration build
Remove rawhide from CI due to issue on jenkins builder
Fixe mypy by installing additional types-* packages

Signed-off-by: Adam Saleh <asaleh@redhat.com>

cverna commented on devel/ci/Dockerfile-rawhide:
```diff
@@ -1,7 +1,10 @@
-FROM registry.fedoraproject.org/fedora:rawhide
+FROM registry.fedoraproject.org/fedora:35
```
This does not seems correct, since rawhide is now f36. We should keep the rawhide label.

cverna commented on devel/ci/integration/bodhi/Dockerfile-f32:
```diff
@@ -1,4 +1,4 @@
-FROM bodhi-ci/f32
+FROM bodhi-ci/f34
```
should probably rename the file too

cverna commented on requirements.txt:
```diff
@@ -28,5 +24,3 @@ PyYAML
 sqlalchemy
 waitress
-WebOb-GraphQL
-whitenoise
```
I think whitenoise is used to serve the docs so we should keep it.

cverna commented on requirements.txt:
```diff
@@ -10,12 +10,8 @@ dogpile.cache
-koji
 jinja2
-markdown>=3.0
-prometheus_client
```
These 2 should probably still be there

cverna commented on requirements.txt:
```diff
@@ -10,12 +10,8 @@ dogpile.cache
 feedgen>=0.7.0
-graphene
-graphene-sqlalchemy
-koji
```
Pretty sure we still want to have koji here :)

humaton committed:
Review changes

Signed-off-by: Tomas Hrcka <thrcka@redhat.com>

mattiaverga commented on devel/ci/integration/bodhi/Dockerfile-f34:
```diff
@@ -1,4 +1,4 @@
-FROM bodhi-ci/f32
+FROM bodhi-ci/f34
```
You will need to change
```
RUN python3 setup.py build && pip3 install --no-use-pep517 . && pip3 install koji
```
with
```
RUN python3 setup.py bdist_wheel && pip3 install ./dist/*.whl && pip3 install koji
```
to fix f34-integration-build

mattiaverga replied:
Oh, also add python3-wheel to dnf install command, see https://github.com/fedora-infra/bodhi/pull/4195/commits/6f5d36debbbca31d7b394308a17217015946766a

AdamSaleh replied:
Done and done. f34-integration-build  works now, thanks!

mattiaverga replied:
You're welcome.

For the f32-* tests which are always required, I think there is a setting at project level in github that can only be changed by an admin... unfortunately, removing them from mergify config isn't enough.

AdamSaleh replied:
Found that setting as well. You think this looks good enough? I'd really like to merge this and then start looking at some of  the stale PR's we have around the place, with the work of fixing and refactoring the integration tests to be postponed to a later date.

AdamSaleh committed:
Change the setup.py command on Mattia's suggestion

Signed-off-by: Adam Saleh <asaleh@redhat.com>

mattiaverga reviewed (approved):
LGTM Thank you!

mattiaverga commented:
@Mergifyio refresh
